aspnetcore/performance/caching/response.md

```diff
@@ -1,92 +1,72 @@
 
-Additional HTTP headers used for caching include `Pragma` and `Vary`, which are described below. Learn more about [Caching in HTTP from the specification](https://tools.ietf.org/html/rfc7234#section-3).
+Caching responses on the web server can be enabled by adding the [response caching middleware](https://github.com/aspnet/responsecaching). See [Response caching middleware](middleware.md) for more information.
 
 ## ResponseCache Attribute
 
-The `ResponseCacheAttribute` is used to specify how a controller action's headers should be set to control its cache behavior. The attribute has the following properties, all of which are optional unless otherwise noted.
+The [ResponseCacheAttribute](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.mvc.responsecacheattribute) specifies the parameters necessary for setting appropriate headers in response caching. See [ResponseCacheAttribute](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.aspnetcore.mvc.responsecacheattribute)  for a description of the parameters.
 
-**Duration `int`**
+`VaryByQueryKeys string[]`: When set, the response caching middleware will vary the stored response by the given list of query keys. The middleware will serve the stored response only if the query keys of the request matches those of the original request that generated the stored response. Setting this property without adding the middleware will throw a runttime exception. 
 
-   The maximum duration (in seconds) the response should be cached. **Required** unless `NoStore` is `true`.
+  There is no corresponding HTTP header for this property. This property is an HTTP feature handled by the response caching middleware. To set the `VaryByQueryKeys` property, the response caching middleware must be enabled.
 
-**Location `ResponseCacheLocation`**
+The `ResponseCacheAttribute` is used to configure and create (via `IFilterFactory`) a `ResponseCacheFilter`. The `ResponseCacheFilter` performs the work of updating the appropriate HTTP headers and features of the response. The filter:
 
-   The location where the response may be cached. May be `Any`, `None`, or `Client`. Default is `Any`.
+* Removes any existing headers for `Vary`, `Cache-Control`, and `Pragma`. 
+* Writes out the appropriate headers based on the properties set in the `ResponseCacheAttribute`. 
+* Updates the response caching HTTP feature if `VaryByQueryKeys` is set.
 
-**NoStore `bool`**
+### `Vary` 
 
-   Determines whether the value should be stored or not, and overrides other property values. When `true`, `Duration` is ignored and `Location` is ignored for values other than `None`.
+This header is only written when the `VaryByHeader` property is set. It is set to the `Vary` property's value. The following sample uses the `VaryByHeader` property.
 
-**VaryByHeader `string`**
+[!code-csharp[Main](response/sample/Controllers/HomeController.cs?name=snippet_VaryByHeader&highlight=1)]
```

Comment:
@JunTaoLuo  the first time I invoke About2 I get the following: (note, not cached)
![capture](https://cloud.githubusercontent.com/assets/3605364/20444221/430c7b62-ad74-11e6-9842-8471c1ac717d.PNG)

The 2nd time I run it (now cached), I get  `User-Agent, Accept-Encoding`